Documents/Installation-with-Conan.md

```diff
@@ -0,0 +1,57 @@
+## Guide to Install and Setup nmos-cpp using Conan Package Manager: 
+
+1. Install conan:
+`pip install --upgrade conan>=2.0.5`
+
+    If conan.exe is not in PATH you will get warning: 
+	 
+
+	> WARNING: The script conan.exe is installed in 'C:\Users\%USER%\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts' which is not on PATH.
+	  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
+
+
+   Then the shown path needs to be added manually. After maunally adding the PATH, Restart the command line and run `conan --help` to ensure Conan is added to path successfully.
+
+3. Detect Profile:
+	Run `conan profile detect`. \
+  This will create a conan profile with values based on your dev setup. Ideally the profile should be similar to these:
+	
+   For windows:
+	```{txt}
+	[settings]
+	arch=x86_64
+	build_type=Release
+	compiler=msvc
+	compiler.cppstd=14
+	compiler.runtime=dynamic
+	compiler.runtime_type=Release
+	compiler.version=193
+	os=Windows
+	```
+	For Linux:
+	```{txt}
+	[settings]
+	arch=x86_64
+	build_type=Release
+	compiler=gcc
+	compiler.cppstd=17
+	compiler.version=11
+	os=Linux
+	```
+
+4. Install nmos cpp:
+	`conan install --tool-requires=nmos-cpp/cci.20240223`\
+  If on windows, also run `./conanbuild.bat` to ensure the executables for nmos-cpp-node and nmos-cpp-registry are added to path.
+
+   Alternatively if the nmos-cpp-node and nmos-cpp-registry has still not been added to path, you can add them manually (Executables likely located at C:\Users\%USERNAME%\.conan2\p\nmos-768c7905bf562\p\bin\Release) or work around this as shown below:
+
+    
+	> Executables can be copied to current working directory using a direct installation:\
+	> `conan install --requires=nmos-cpp/cci.20240223 --deployer=direct_deploy`
+   	> - `--deployer=direct_deploy` copies the direct dependencies, in this case nmos-cpp-registry.exe and nmos-cpp-node.exe.
+   	>
+ 	> \
+	> Launch the registry using `./direct_deploy/nmos-cpp/bin/Release/nmos-cpp-registry.exe <your_config.json>` \
```

Comment:
Oddly the documentation implies that the path includes the version as well as the dependency name, like **_direct_deploy/nmos-cpp/cci.20240223/bin/Release/nmos-cpp-registry_**.

Is that not the case?

https://docs.conan.io/2.0/reference/extensions/deployers.html#direct-deploy

Reply:
![On my installation on windows](https://github.com/johanjino/nmos-cpp/assets/92362159/8b52a28b-b3dd-430b-a21d-877c9f8e5703)

This was how it was installed for me in Windows. Tried other packages are all of them did not have version. I later went to [conan installation script](https://github.com/conan-io/conan/blob/develop2/conan/internal/deploy.py#L118) and found this: 

```python
_deploy_single(dep, conanfile, output_folder, dep.ref.name)
```
Direct deploy only creates a folder with the ref.name, which here is nmos-cpp. On the other hand:

```python
folder_name = os.path.join("full_deploy", dep.context, dep.ref.name, str(dep.ref.version))
...
 _deploy_single(dep, conanfile, output_folder, folder_name)
```
Full deploy uses ref.version as well to generate folder name.


Seems like either a bug or an error in doc. Should I point this out as an GitHub issue?

Reply:
Good research. Please open an [issue](https://github.com/conan-io/conan/issues) with the references to _deploy.py_ you included above, contrasted to the docs at https://github.com/conan-io/docs/blob/1757da950c00155b3410d9bcd1cb815a41f899e4/reference/extensions/deployers.rst?plain=1#L51.

Reply:
I think it's also instructive to read @memsharded guidance on virtual envs vs direct deploy - https://github.com/conan-io/conan/issues/15913#issuecomment-2014944542

Reply:
I'd say some small difference among the deployers are expected:
- The ``direct_deploy`` is intended to deploy one package, without having to deploy its dependencies. Mostly for convenient extraction of some files inside a given package. So the files won't collide with other package files when deployed. The pkg name has been respected, but it could have been dropped too.
- The ``full_deploy`` deploys the full dependency graph. It is common that more than 1 different version of the same dependency occurs on a graph, then the version becomes more necessary to avoid collisions in the deployment.

Reply:
@memsharded Makes sense, thanks! In which case, perhaps just the example in the documentation for the direct deployer is slightly wrong, since it includes a version directory.

Reply:
Could be outdated or wrong, yep. If you want to do it, it would be great if you can submit a ticket or a PR to the docs about it, otherwise I can do it myself. Thanks for pointing this out!

Reply:
Done: https://github.com/conan-io/docs/pull/3675